src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1982,76 +1960,82 @@ namespace cryptonote
-      LOKI_DEFER { m_miner.resume(); };
-      try
-      {
+      try {
         blocks.push_back(get_block_complete_entry(b, m_mempool));
       }
-      catch (const std::exception &e)
-      {
+      catch (const std::exception &e) {
         return false;
       }
       std::vector<block> pblocks;
-      if (!prepare_handle_incoming_blocks(blocks, pblocks))
-      {
+      if (!prepare_handle_incoming_blocks(blocks, pblocks)) {
         MERROR("Block found, but failed to prepare to add");
         return false;
       }
       add_new_block(b, bvc, nullptr /*checkpoint*/);
       cleanup_handle_incoming_blocks(true);
-      m_miner.on_block_chain_update();
     }
 
     CHECK_AND_ASSERT_MES(!bvc.m_verifivation_failed, false, "mined block failed verification");
-    if(bvc.m_added_to_main_chain)
-    {
+    if (bvc.m_added_to_main_chain) {
       std::vector<crypto::hash> missed_txs;
       std::vector<cryptonote::blobdata> txs;
       m_blockchain_storage.get_transactions_blobs(b.tx_hashes, txs, missed_txs);
-      if(missed_txs.size() &&  m_blockchain_storage.get_block_id_by_height(get_block_height(b)) != get_block_hash(b))
-      {
+      if (missed_txs.size() && m_blockchain_storage.get_block_id_by_height(get_block_height(b)) != get_block_hash(b)) {
         LOG_PRINT_L1("Block found but, seems that reorganize just happened after that, do not relay this block");
         return true;
       }
-      CHECK_AND_ASSERT_MES(txs.size() == b.tx_hashes.size() && !missed_txs.size(), false, "can't find some transactions in found block:" << get_block_hash(b) << " txs.size()=" << txs.size()
-        << ", b.tx_hashes.size()=" << b.tx_hashes.size() << ", missed_txs.size()" << missed_txs.size());
-
+      CHECK_AND_ASSERT_MES(txs.size() == b.tx_hashes.size() && !missed_txs.size(), false,
+                           "can't find some transactions in found block:" << get_block_hash(b) << " txs.size()="
+                                                                          << txs.size()
+                                                                          << ", b.tx_hashes.size()="
+                                                                          << b.tx_hashes.size() << ", missed_txs.size()"
+                                                                          << missed_txs.size());
       cryptonote_connection_context exclude_context{};
       NOTIFY_NEW_FLUFFY_BLOCK::request arg{};
-      arg.current_blockchain_height                 = m_blockchain_storage.get_current_blockchain_height();
-      arg.b                                         = blocks[0];
+
+      arg.current_blockchain_height = m_blockchain_storage.get_current_blockchain_height();
+      arg.b = blocks[0];
 
       m_pprotocol->relay_block(arg, exclude_context);
     }
     return true;
   }
-  //-----------------------------------------------------------------------------------------------
-  void core::on_synchronized()
-  {
-    m_miner.on_synchronized();
```

Comment:
This handles pausing the miner until the blockchain is ready which is desirable behaviour, but not handled anymore?